packages/dds/shared-object-base/src/sharedObject.ts

```diff
@@ -774,3 +775,51 @@ export abstract class SharedObject<
 		telemetryContext?.set(this.telemetryContextPrefix, propertyName, prevTotal + incrementBy);
 	}
 }
+
+/**
+ * Defines a kind of shared object.
+ * Used in containers to register a shared object implementation, and to create new instances of a given type of shared object.
+ * @public
+ */
+export interface ISharedObjectKind<TSharedObject> {
+	/**
+	 * Get a factory which can be used by the Fluid Framework to programmatically instantiate shared objects within containers.
+	 * @remarks
+	 * The produced factory is intended for use with the FluidDataStoreRegistry and is used by the Fluid Framework to instantiate already existing Channels.
+	 * To create new shared objects use:
+	 *
+	 * - {@link @fluidframework/fluid-static#IFluidContainer.create} if using `@fluidframework/fluid-static`, for example via `@fluidframework/azure-client`.
+	 * - {@link ISharedObjectKind.create} if using a custom container definitions (and thus not using {@link @fluidframework/fluid-static#IFluidContainer}).
```

Comment:
Nit (to preserve the list-ness in the generated docs, until we support markdown parsing)
```suggestion
	 *
	 * - {@link ISharedObjectKind.create} if using a custom container definitions (and thus not using {@link @fluidframework/fluid-static#IFluidContainer}).
```